src/tableFormatter.ts

```diff
@@ -126,13 +140,12 @@ class MarkdownDocumentFormatter implements DocumentFormattingEditProvider {
             while ((field = fieldRegExp.exec(row)) !== null) {
                 let cell = field[1].trim();
                 values.push(cell);
-
                 //// Calculate `colWidth`
-                //// Ignore length of dash-line to enable width reduction
-                if (num != 1) {
+                //// Ignore length of delimiter-line to enable width reduction
+                if (num != delimiterRowNum) {
                     //// Treat CJK characters as 2 English ones because of Unicode stuff
                     const numOfUnicodeChars = splitter.countGraphemes(cell);
-                    const width = cjkRegex.test(cell) ? numOfUnicodeChars + cell.match(cjkRegex).length : numOfUnicodeChars;
+                    const width = (cjkRegex.test(cell) ? numOfUnicodeChars + cell.match(cjkRegex).length : numOfUnicodeChars) + cellPaddingLength;
```

Comment:
I would prefer not to change the meaning from content width to padded width here. (see below)

Reply:
See my reaction to your last comment why I did it this way.